core/integration/client_tests.go

```diff
@@ -492,3 +501,172 @@ func (m uint64Slice) Less(i, j int) bool { return m[i] < m[j] }
 func cp(tag int) []byte {
 	return []byte(fmt.Sprintf("bar%v", tag))
 }
+
+// GenerateTestVectors verifies set/get semantics.
+// TODO(gbelvin): Migrate into core/integration/client_tests to avoid code rot.
+func GenerateTestVectors(ctx context.Context, env *Env) error {
+	go func() {
+		ticker := time.NewTicker(100 * time.Millisecond)
+		defer ticker.Stop()
+		sequencer.PeriodicallyRun(ctx, ticker.C, func(ctx context.Context) {
+			req := &spb.RunBatchRequest{
+				DirectoryId: env.Directory.DirectoryId,
+				MinBatch:    1,
+				MaxBatch:    100,
+			}
+			if _, err := env.Sequencer.RunBatch(ctx, req); err != nil {
+				log.Errorf("RunBatch(): %v", err)
+			}
+		})
+	}()
+	testdataDir := "."
+	// Create lists of signers.
+	signers1 := testutil.SignKeysetsFromPEMs(testPrivKey1)
+
+	// Create lists of authorized keys
+	authorizedKeys1 := testutil.VerifyKeysetFromPEMs(testPubKey1).Keyset()
+
+	// Collect a list of valid GetUserResponses
+	getUserResps := make([]testdata.GetUserResponseVector, 0)
+
+	// Start with an empty trusted log root
+	slr := &types.LogRootV1{}
+
+	for _, tc := range []struct {
+		desc           string
+		wantProfile    []byte
+		setProfile     []byte
+		ctx            context.Context
+		userID         string
+		signers        []tink.Signer
+		authorizedKeys *tinkpb.Keyset
+	}{
+		{
+			desc:           "empty_alice",
+			wantProfile:    nil,
+			setProfile:     []byte("alice-key1"),
+			ctx:            authentication.WithOutgoingFakeAuth(ctx, "alice"),
```

Comment:
`TestEmptyGetAndUpdate` uses the following to avoid an import reference to `impl`.
```suggestion
			opts:           env.CallOpts("alice"),
```